### DISCUSSION TITLE
Improve PR body truncation

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#21669] [Suggest an Idea]
If release notes are too long, then we truncate them arbitrarily. Ideally we could do this more cleverly, especially when there are multiple packages in a PR.

**Comment by pi0** [comment]
This is in the continuation of https://github.com/renovatebot/presets/issues/85#issuecomment-425134563.

> The babel release notes are huge and meanwhile, there's a limit to how many characters you can fit in a GitHub PR body, so a lot had to be truncated. Notice there's no "Created by Renovate Bot" footer there either. Anyway right now we do the simplest approach and simply truncate PR body text starting from the beginning, but in this case, it's meant only babel can fit.

My suggested behaviour is to extract big changelogs into additional replies and maintain them in case of edits.

**Create a PR:**

- 1. Gather changelogs
- 2. Scan through and change the text of longer ones intro a short message/link like "See below"
- 3. For each excluded item in step (ii), add an additional message with an special HTML comment

**Modify a PR:** 

- 1. Do steps (i) and (ii)
- 2. Read all previous replies for special HTML comment
- 3. Do step (iii) but only edit posts if already added in an additional reply

Now we can also truncate _big_ replies and add a link to the original changelog.

**Comment by rarkins** [comment]
@pi0 thanks for the ideas. Indeed, *expanding* the existing PR body to extend it into 1 or more additional "comments" is an alternative to truncating. In fact at one point I had considered whether to move *all* embedded changelogs out of the main PR body for those who like to merge squash.

**Comment by rarkins** [comment]
I've also added a small tweak just now to make sure we don't truncate the PR's configuration and footer and instead only truncate the "middle" release notes section.

**Comment by viceice** [comment]
Moving changelogs to comments is a good idea, like that.
